File: Study Plan/Top Interview 150/28 Find the Index of the First Occurrence in a String/FindTheIndexOfTheFirstOccurrenceInAString.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    28. Find the Index of the First Occurrence in a String
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/solutions/4384285/c-time-complexity-o-m-n-space-complexity-o-1-beats-71-in-runtime/

    - Time Complexity O(m * n)
    - Space Complexity O(1)

    - Runtime 56 ms (Beats 71.05% of users with C#)
    - Memory 37.56 MB (Beats 9.55% of users with C#)
*/
public class Solution
{
    public int StrStr(string haystack, string needle)
    {
        if (haystack.Contains(needle))
        {
            return haystack.IndexOf(needle);
        }
        else
        {
            return -1;
        }
    }
}

File: Study Plan/Top Interview 150/28 Find the Index of the First Occurrence in a String/FindTheIndexOfTheFirstOccurrenceInAString.java
/*  
    Problem
    --------------------------------------------------------------------------------
    28. Find the Index of the First Occurrence in a String
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/solutions/4432832/clean-simple-solution-in-java-beats-100-in-runtime/

    - Runtime 0 ms (Beats 100.00% of users with Java)
    - Memory 41.36 MB (Beats 5.11% of users with Java)
*/

class Solution {
    public int strStr(String haystack, String needle) {
        if (haystack.contains(needle)) {
            return haystack.indexOf(needle);
        } else {
            return -1;
        }
    }
}
